hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -265,6 +268,23 @@ public ContainerStateMachine(HddsDatanodeService hddsDatanodeService, RaftGroupI
 
   }
 
+  private void validatePeers(RaftServer server, RaftGroupId id) throws IOException {
+    if (this.peersValidated.get()) {
+      return;
+    }
+    RaftPeerId selfId = server.getId();
+    Collection<RaftPeer> peers = server.getDivision(id).getGroup().getPeers();
+    // If peers list is empty then it means  Ratis hasn't created any raft--meta file containing the last applied
+    // transaction. Then the peer list can be only validated on apply transaction.
+    if (!peers.isEmpty() && peers.stream().noneMatch(raftPeer -> raftPeer != null
+        && raftPeer.getId().equals(selfId))) {
+      throw new StorageContainerException(String.format("Current datanodeId: %s is not part of the " +
+          "group : %s with quorum: %s", selfId, id, peers), ContainerProtos.Result.INVALID_CONFIG);
+    } else if (!peers.isEmpty()) {
+      peersValidated.set(true);
+    }
```

Comment:
This can be simplified as below:
```java
  private void validatePeers() throws IOException {
    if (peersValidated.get()) {
      return;
    }
    final RaftGroup group = ratisServer.getServerDivision(getGroupId()).getGroup();
    final RaftPeerId selfId = ratisServer.getServer().getId();
    if (group.getPeer(selfId) == null) {
      throw new StorageContainerException("Current datanode " + selfId + " is not a member of " + group,
          ContainerProtos.Result.INVALID_CONFIG);
    }
    peersValidated.set(true);
  }
```

Reply:
done